src/main/java/cubicchunks/server/ServerCubeCache.java

```diff
@@ -269,151 +263,156 @@ public boolean chunkExists(int cubeX, int cubeZ) {
 	//==============================
 
 	@Override
-	public boolean cubeExists(int cubeX, int cubeY, int cubeZ) {
-		long columnAddress = AddressTools.getAddress(cubeX, cubeZ);
-		Column column = this.loadedColumns.get(columnAddress);
-		if (column == null || column.unloaded) {
-			return false;
-		}
-		Cube cube = column.getCube(cubeY);
-		return cube != null && !cube.unloaded;
+	public Cube getCube(int cubeX, int cubeY, int cubeZ) {
+		return getCube(new CubeCoords(cubeX, cubeY, cubeZ));
```

Comment:
I think it would make sense to use the long-encoded address as keys to a LongMap here